src/main/kotlin/StringAddCalculator.kt

```diff
@@ -0,0 +1,54 @@
+import java.lang.RuntimeException
+
+class StringAddCalculator(strExpression: String?) {
+    var result: Int = 0
+    var arrNumber: List<String> = ArrayList()
+
+    init {
+        if (strExpression.isNullOrEmpty())
+            result = 0
+        else {
+            checkCustomDelimiter(strExpression)
+        }
+    }
+
+    fun checkCustomDelimiter(strExpression: String) {
+        val check = Regex("//(.)\n(.*)").find(strExpression)
+        if (check != null) {
+            check.let {
+                val customDelimiter = it.groupValues[1]
+                arrNumber = it.groupValues[2].split(customDelimiter)
+                amountArrNumber()
+            }
```

Comment:
if 문의 조건식에서 null이 아님을 체크함으로써 [smart cast](https://bbaktaeho-95.tistory.com/24)가 됩니다. 😃
따라서 if문 내부에서는 check는 nullable하지 않은 값이 되기 때문에 `let` scope 함수를 쓸 필요가 없어요. 🙂

```suggestion
            val customDelimiter = check.groupValues[1]
            arrNumber = check.groupValues[2].split(customDelimiter)
            amountArrNumber()
```